tests/conftest.py

```
@@ -0,0 +1,78 @@
+"""Global pytest configuration and project-wide fixtures."""
+
+from unittest.mock import MagicMock
+
+import pytest
+
+
+class DummySessionState(dict):
+    """Mock session state that supports both dict and attribute access."""
+
+    def __getattr__(self, key):
```

Ruff (0.15.12)

[warning] Missing return type annotation for special method `__getattr__`

(ANN204)